model/schema/highThroughputExpression.yaml

```diff
@@ -0,0 +1,578 @@
+id: https://github.com/alliance-genome/agr_persistent_schema/src/schema/highThroughputExpression.yaml
+name: High-Throughput-Expression-Dataset-Annotation
+title: High Throughput Expression Dataset Annotation
+description: Annotations for high-throughput gene expression experiments and datasets
+
+imports:
+  - core
+  - expression
+  - linkml:types
+
+prefixes:
+  alliance: 'http://alliancegenome.org/'
+  linkml: 'https://w3id.org/linkml/'
+  gff: 'https://w3id.org/gff'
+  faldo: 'http://biohackathon.org/resource/faldo#'
+  biolink: 'https://w3id.org/biolink/vocab/'
+  NLMID: 'https://www.ncbi.nlm.nih.gov/nlmcatalog/?term='
+  schema: 'http://schema.org/'
+  dct: 'http://purl.org/dc/terms/'
+  WIKIDATA_PROPERTY: 'https://www.wikidata.org/wiki/Property:'
+  obograph: 'https://github.com/biodatamodels/obograph'
+
+default_prefix: alliance
+default_range: string
+
+default_curi_maps:
+  - obo_context
+  - idot_context
+  - semweb_context
+  - monarch_context
+
+emit_prefixes:
+  - rdf
+  - rdfs
+  - xsd
+  - owl
+
+
+classes:
+
+  HighThroughputExpressionDatasetAnnotation:
+    is_a: SubmittedObject
+    description: >-
+      An entry for a published high-throughput expression dataset annotation.
+    slots:
+      - dataset_id # datasetId in agr_schemas json
```

Comment:
Do we need this?  SubmittedObject already has `mod_entity_id` and `mod_internal_id`.

Reply:
@markquintontulloch Yeah, so I debated with myself on this one. Generally, these are IDs for 3rd party databases, like GEO and ArrayExpress. Here's the slot definition description: 

"The ID from the data provider for the dataset. When available, this is the GEO ID. When no GEO ID is available, it will be the ArrayExpress ID. When these both are unavailable, it will be the MOD ID."

so, occasionally it will be a MOD identifier, but usually not. Do we really want to use 'mod_entity_id' for these? Or maybe we need to repurpose mod_entity_id to be db_entity_id to apply more generally? I just feel a bit odd trying to call these MOD IDs

Reply:
I'd be in favour of renaming and repurposing `mod_entity_id` to be more general - I think we're already kind of stretching the definition in places already.  The other option here would be to inherit from `CurieObject` instead of `SubmittedObject` and add `data_provider` and `database_id` slots to this class.  As it is, it feels like `mod_entity_id` and `mod_internal_id` are never going to be used for this class, unless that's likely to change once we eventually move to direct submissions.